src/cli/action.zig

```diff
@@ -68,17 +76,33 @@ pub const Action = enum {
     pub const help_error = error.ActionHelpRequested;
 
     /// Detect the action from CLI args.
-    pub fn detectCLI(alloc: Allocator) !?Action {
+    /// Pass in a pointer to a LoggingOptions structure if you want to retrieve
+    /// any `-q/--quiet` or `-v/--verbose` options or `null` if you don't need them.
+    pub fn detectCLI(alloc: Allocator, loggingOptions: ?*LoggingOptions) !?Action {
         var iter = try std.process.argsWithAllocator(alloc);
         defer iter.deinit();
-        return try detectIter(&iter);
+        return try detectIter(&iter, loggingOptions);
```

Comment:
Out-pointers are generally somewhat frowned upon in the codebase, but I'm not exactly sure what's a better alternative

Reply:
Searching revealed out-pointers were the most recommended alternative.

There were a couple of other options I tested...
* Change Action from an enum to a struct so when it's returned to global it can contain more than just the action.  That's a fairly invasive change however because other modules like bash_completion.zig would need to be changed.
* Store the results of the parsing in a top-level cli.action structure and add a getter for it.  Kinda clunky.
* Don't use the existing iterator in Action and add one in main_ghostty just to get the logging options.  Worked fine but I thought using the existing iterator would save a bit.

Reply:
I added a quick arguments iterator in global just before the call to Action.detectCLI().  This removes the need to pass the pointer.